packages/ketcher-core/src/application/render/draw.ts

```diff
@@ -144,636 +215,745 @@ function arrow(
-        point1.y + arrowOffset
+    `M${tfx(start.x + unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `L${tfx(endX - unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `M${tfx(start.x + unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `L${tfx(start.x + arrowLen + unbalanceVal)},${tfx(
+        start.y + arrowOffset
       )}` +
-      `L${tfx(point1.x + arrowLen - arrowHeadAttr + unbalanceVal)},${
-        point1.y + lineOffset
+      `L${tfx(start.x + arrowLen - arrowHeadAttr + unbalanceVal)},${
+        start.y + lineOffset
       }Z`
   )
 
   const transformedPath = svgPath(path.join(''))
-    .rotate(arrowAngle, point1.x, point1.y)
+    .rotate(arrowAngle, start.x, start.y)
     .toString()
 
-  return paper.path(transformedPath).attr({ ...options.lineattr, fill: '#000' })
+  return paper.path(transformedPath).attr({
+    ...options.lineattr,
+    fill: '#000',
+    ...(shouldApplySnappingStyle && options.arrowSnappingStyle)
+  })
 }
 
 function arrowUnbalancedEquilibriumFilledHalfTriangle(
   paper: RaphaelPaper,
-  { pos: [point1] }: ArrowItem,
+  { pos: [start] }: ArrowItem,
   arrowLength: number,
   arrowAngle: number,
-  options: RenderOptions
+  options: RenderOptions,
+  shouldApplySnappingStyle: boolean
 ) {
   const arrowLen = 9
   const lineOffset = 3.5
   const arrowOffset = 7
   const unbalanceVal = 15
 
-  const b0x = point1.x + arrowLength
+  const endX = start.x + arrowLength
 
   const path: string[] = []
 
   // First arrow
   path.push(
-    `M${tfx(point1.x)},${tfx(point1.y - lineOffset)}` +
-      `L${tfx(b0x)},${tfx(point1.y - lineOffset)}` +
-      `L${tfx(b0x - arrowLen)},${tfx(point1.y - arrowOffset)}` +
-      `L${tfx(b0x - arrowLen)},${tfx(point1.y - lineOffset)}Z`
+    `M${tfx(start.x)},${tfx(start.y - lineOffset)}` +
+      `L${tfx(endX)},${tfx(start.y - lineOffset)}` +
+      `L${tfx(endX - arrowLen)},${tfx(start.y - arrowOffset)}` +
+      `L${tfx(endX - arrowLen)},${tfx(start.y - lineOffset)}Z`
   )
 
   // Second (Unbalanced) arrow
   path.push(
-    `M${tfx(point1.x + unbalanceVal)},${tfx(point1.y + lineOffset)}` +
-      `L${tfx(b0x - unbalanceVal)},${tfx(point1.y + lineOffset)}` +
-      `M${tfx(point1.x + unbalanceVal)},${tfx(point1.y + lineOffset)}` +
-      `L${tfx(point1.x + arrowLen + unbalanceVal)},${tfx(
-        point1.y + arrowOffset
+    `M${tfx(start.x + unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `L${tfx(endX - unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `M${tfx(start.x + unbalanceVal)},${tfx(start.y + lineOffset)}` +
+      `L${tfx(start.x + arrowLen + unbalanceVal)},${tfx(
+        start.y + arrowOffset
       )}` +
-      `L${tfx(point1.x + arrowLen + unbalanceVal)},${point1.y + lineOffset}Z`
+      `L${tfx(start.x + arrowLen + unbalanceVal)},${start.y + lineOffset}Z`
   )
 
   const transformedPath = svgPath(path.join(''))
-    .rotate(arrowAngle, point1.x, point1.y)
+    .rotate(arrowAngle, start.x, start.y)
```

Comment:
suggestion for the next PR: could you create Refactoring PR separately - it takes time to review each line and distinguesh - what was the function change and what was the rename?

Reply:
Sorry for mixing the new feature and refactoring together. Will pay more attention next time, thank you!

Reply:
Or other proposal – you can squash commits and for example, create 2: one for the task itself and another one for refactoring

Reply:
Good proposal, thank you! Done